thentos-core/src/Thentos/Util.hs

```diff
@@ -86,16 +83,6 @@ readsPrecEnumBoundedShow _ s = f [minBound..]
         s' = show x
 
 
--- | Path concatenation for avoiding double slashes in paths.  One
```

Comment:
I created the `stripLeadingSlash` and `stripTrailingSlash` helper functions because I need one of them in `renderProxyUri`. I cannot import Util into Types without an import loop and it felt strange to have the helpers in one module and the operator using them in another, hence I moved everything into Types.

As for the INLINE pragma, I would tend to let the compiler worry about that, except in cases where we found a performance bottleneck via benchmarking. Also, is there even a performance difference between a top-level definition and one defined in a where clause?